pkg/kubecost/coverage.go

```diff
@@ -0,0 +1,108 @@
+package kubecost
+
+import (
+	"time"
+
+	"github.com/opencost/opencost/pkg/filter"
+)
+
+// Coverage This is a placeholder struct which can be replaced by a more specific implementation later
+type Coverage struct {
+	Window   Window    `json:"window"`
+	Type     string    `json:"type"`
+	Count    int       `json:"count"`
+	Updated  time.Time `json:"updated"`
+	Errors   []string  `json:"errors"`
+	Warnings []string  `json:"warnings"`
+}
+
+func (c *Coverage) GetWindow() Window {
+	return c.Window
+}
+
+func (c *Coverage) Key() string {
+	return c.Type
+}
+
+func (c *Coverage) IsEmpty() bool {
+	return c.Type == "" && c.Count == 0 && len(c.Errors) == 0 && len(c.Warnings) == 0 && c.Updated == time.Time{}
+}
+
+func (c *Coverage) Clone() *Coverage {
+	return &Coverage{
+		Window:   c.Window.Clone(),
+		Type:     c.Type,
+		Count:    c.Count,
+		Updated:  c.Updated,
+		Errors:   append(c.Errors),
+		Warnings: append(c.Warnings),
```

Comment:
I don't think this `append` usage is a true copy: https://go.dev/play/p/xoieGEkcNxP

Reply:
It's possible you could use something like `append([]string{}, c.Errors...)` instead. However, the following is preferable:

```go
var errors []string 
if len(c.Errors) > 0 {
    errors = make([]string, len(c.Errors))
    copy(errors, c.Errors)
}
```